lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -379,7 +380,9 @@ bool SPIRVRegularizeLLVMBase::regularize() {
         if (auto Call = dyn_cast<CallInst>(&II)) {
           Call->setTailCall(false);
           Function *CF = Call->getCalledFunction();
-          if (CF && CF->isIntrinsic()) {
+          if (CF && CF->getCallingConv() == CallingConv::SPIR_KERNEL) {
+            CalledKernels.push_back(CF);
```

Comment:
I assume this doesn't detect kernels launched through `enqueue_kernel` (aka device execution/device enqueue).  Do we need to handle that here too?

Reply:
Sounds reasonable, thanks for checking!